primitives/src/lib.rs

```diff
@@ -506,13 +507,25 @@ pub enum ForeignCurrencyId {
 	USD,   // Statemine
 }
 
+impl ForeignCurrencyId {
+	pub fn decimal(&self) -> u8 {
+		10
```

Comment:
Can we assume this is always the case? Can we just set it to any value we like?

Reply:
so, instead of doing all of these all over again, i just replaced `TokenSymbol` with `ForeignCurrencyId`. 
And I think it works